src/Windows/Avalonia.Win32/WinRT/Composition/WinUiCompositedWindowSurface.cs

```diff
@@ -83,28 +79,33 @@ internal class WinUiCompositedWindowRenderTarget : IDirect3D11TextureRenderTarge
                 _surface = _drawingSurface.QueryInterface<ICompositionSurface>();
                 _surfaceInterop = _drawingSurface.QueryInterface<ICompositionDrawingSurfaceInterop>();
             }
-            finally
+            catch
```

Comment:
I'm not sure that copy-pasting stuff is a good idea, especially the `Dispose` code that deals with unmanaged resources

Reply:
I'm not a fan of duplication either, but I found it a good compromise here:
 - Making those fields nullable means plastering `!` on every use: in this case the code is quite short and understandable so it should be fine. In my experience it tends to simply hide real NRE when overused, especially in future modifications or when the code become more complex: you don't really know if the variable is supposed to be null or not: the annotations lose their purposes.
 - Making those fields non nullable reflects the reality: but calling Dispose means having to keep the null propagation syntax `field?.Dispose`, then you wonder why since it's supposed to be non nullable.

I really like trusting annotations and contracts in general (not only for nullability): I find it really helps reading code when you don't have to guess too much what the state of everything is before calling some method. That said, that's your codebase, if you tell me I should revert those I will, there's no problem! I'm just trying to explain my reasoning here :)